Mathlib/Data/List/Basic.lean

```diff
@@ -395,6 +395,11 @@ theorem cons_eq_append_iff {a b c : List α} {x : α} :
 @[deprecated] alias append_right_cancel := append_cancel_right -- deprecated since 2024-01-18
 #align list.append_right_cancel List.append_cancel_right
 
+@[simp] theorem append_left_cancel_nil_left {x y : List α}: x ++ y = x ↔ y = [] := by
```

Comment:
A few comments:
- please put spaces around the colon, i.e. `{x y : List α} : x ++ y = x ↔ y = []`
- in general, writing several tactics on one line is frowned upon (there are exceptions, but this is a good rule of thumb)

Altogether, I'd write this more like
```lean
@[simp]
theorem append_left_cancel_nil_left {x y : List α} : x ++ y = x ↔ y = [] := by
  constructor <;>
  intro h <;>
  apply List.append_cancel_left (as := x) <;>
  all_goals simp [h]
```